### PR TITLE
14-rest-api-code-refactoring

### LINKED ISSUE #14: [rest-api] Code refactoring
1. Find out how the app works
2. Try to run the application, you can use the script https://github.com/Kv-126-DevOps/None/issues/15#issuecomment-1079721724
3. In your opinion, add the necessary changes to the code
    - Format code according to PEP-8
    - Add check of required environment variables; set other environment variables to default values
    - Rename the environment variables according to the accepted standard naming convention
    - Fill out README.md according to the [README_service_template.md](https://github.com/Kv-126-DevOps/None/blob/main/README_service_template.md)
4. Create a pull request from the branch `issue-<ISSUE NUMBER>`

### PR DESCRIPTION
**Pull request by AlexPokatilov**
1. Checked all files for code formating PEP-8
2. Added working documentation
3. Tested the functionality of the service (with another services)

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,5 +1,5 @@
-POSTGRES_HOST=localhost
```
Comment (anatolek) [inline .env.example:1]: Not sure that this file is needed, so I guess it is better just add this info to the README.md and then remove this file at all

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
-FROM python:3.8-slim-buster
+FROM python:3.9-slim
 WORKDIR /usr/src/app
 COPY requirements_old.txt ./
 RUN pip3 install --upgrade pip
```
Comment (anatolek) [inline Dockerfile]: I think it's redundant and can be removed

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
 RUN pip3 install --no-cache-dir -r requirements.txt
```
Comment (anatolek) [inline Dockerfile]: I see the problem here is that we copy `requirements_old.txt` and then use `requirements.txt`

File: Dockerfile [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
 COPY . .
```
Comment (anatolek) [inline Dockerfile]: It would be nice to add a separate non-privileged user here and run the application on his behalf

File: config.py [hunk 5]
```diff
@@ -17,10 +17,12 @@ def getEnvVariable(name):
 POSTGRES_HOST = getEnvVariable("POSTGRES_HOST")
 POSTGRES_PORT = getEnvVariable("POSTGRES_PORT")
```
Comment (anatolek) [inline config.py]: I guess it's better to use something like this and add default value
```python
POSTGRES_PORT = os.environ.get('POSTGRES_PORT', 5432)
```

**Review by anatolek — CHANGES_REQUESTED** [review]

File: .env.example (hunk 1 above)
Comment (AlexPokatilov) [reply .env.example:1]: Ok

File: Dockerfile (hunk 2 above)
Comment (AlexPokatilov) [reply Dockerfile]: Ok

File: config.py (hunk 5 above)
Comment (AlexPokatilov) [reply config.py]: replaced by :
`POSTGRES_PORT = os.environ.get('POSTGRES_PORT', 5432)`
`POSTGRES_PORT = os.environ.get('POSTGRES_DB', "postgres")`

File: Dockerfile (hunk 3 above)
Comment (AlexPokatilov) [reply Dockerfile]: Ok

File: Dockerfile (hunk 4 above)
Comment (AlexPokatilov) [reply Dockerfile]: Tried to fix it in a new commit, but perhaps an example is needed

Comment (anatolek) [reply Dockerfile]: https://medium.com/@mccode/processes-in-containers-should-not-run-as-root-2feae3f0df3b

https://ralph.blog.imixs.com/2017/04/23/run-a-docker-container-with-non-privileged-user/

Comment (AlexPokatilov) [reply Dockerfile]: THX. Checked and added changes

**Pull request merged**